app/jobs/cancel_outdated_reservations_job.rb

```diff
@@ -0,0 +1,27 @@
+class CancelOutdatedReservationsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    reservations.each do |reservation|
+      cancel_reservation(reservation).call unless outdated?(reservation)
+    end
+  end
+
+  private
+
+  def status
+    Reservation::CANCELLED
+  end
+
+  def reservations
+    Reservation.where(status: Reservation::CREATED)
+  end
+
+  def cancel_reservation(reservation)
+    Reservations::UseCases::Update.new(reservation: reservation, status: status)
+  end
+
+  def outdated?(reservation)
+    reservation.seance.start_time >= DateTime.now + Reservation::CONFIRMATION_DEADLINE
```

Comment:
I think it should look a little bit different:

You basically need to check if it is already 30 minutes before the seance.

```ruby
DateTime.now >= reservation.seance.start_time - Reservation::CONFIRMATION_DEADLINE
```

Reply:
I think that in your version it should be rather 
`DateTime.now <= reservation.seance.start_time - Reservation::CONFIRMATION_DEADLINE`

Reply:
Things like that can be quite confusing and difficult to reason about, so to make it clearer I decided to draw a picture, hope it helps. 

![Screenshot 2022-10-06 at 12 18 53](https://user-images.githubusercontent.com/113047354/194289021-40a2be51-6430-472a-a83e-bdaacc7cdc52.png)